src/components/FullscreenGrid/index.tsx

```diff
@@ -3,38 +3,63 @@ import makeClassName from 'classnames';
 
 import styles from './styles.module.scss';
 
-type PanelProps = {
-  children: React.ReactNode | React.ReactNode[];
-  className?: string;
-};
+type AnyReactNode = React.ReactNode | React.ReactNode[];
 
-export const Header = ({ children, className }: PanelProps) => {
-  return (
-    <div className={makeClassName(styles.Header, className)}>{children}</div>
-  );
-};
-
-export const NavPanel = ({ children, className }: PanelProps) => {
+export const Header = ({
+  children,
+  className,
+}: {
+  children?: AnyReactNode;
+  className?: string;
+}) => {
   return (
-    <div className={makeClassName(styles.NavPanel, className)}>{children}</div>
+    <header className={makeClassName(styles.Header, className)}>
+      {children}
+    </header>
   );
 };
 
-export const AltPanel = ({ children, className }: PanelProps) => {
-  return (
-    <div className={makeClassName(styles.AltPanel, className)}>{children}</div>
-  );
+type ContentShellProps = {
+  altSidePanel?: AnyReactNode;
+  altSidePanelClass?: string;
+  children?: AnyReactNode;
+  className?: string;
+  mainSidePanel?: AnyReactNode;
+  mainSidePanelClass?: string;
 };
 
-export const ContentPanel = ({ children, className }: PanelProps) => {
+export const ContentShell = ({
+  altSidePanel,
+  altSidePanelClass,
+  children,
+  className,
+  mainSidePanel,
+  mainSidePanelClass,
+}: ContentShellProps) => {
   return (
-    <div className={makeClassName(styles.ContentPanel, className)}>
-      {children}
-    </div>
+    <React.Fragment>
+      <aside
+        className={makeClassName(styles.mainSidePanel, mainSidePanelClass)}
+      >
+        {mainSidePanel}
+      </aside>
+      <main className={makeClassName(styles.content, className)}>
+        {children}
+      </main>
+      <aside className={makeClassName(styles.altSidePanel, altSidePanelClass)}>
+        {altSidePanel}
```

Comment:
So if `altSidePanel` is undefined then it would show an empty panel? There's no current way to hide the `aside` for this `altSidePanel`?